### ISSUE TITLE
Allow HAML support

### ISSUE THREAD

**Issue by apolishch** [half-pipe/half-pipe#46]

**Comment by joefiorini** [comment]
Can you please explain what you mean by that? Where would you like to use Haml that you currently can't?

**Comment by apolishch** [comment]
I am referring to allowing an app/views/layouts/application.html.haml rather then an app/views/layouts/application.html.erb.

At present if the entire view structure (including application.html) is in haml rather then erb, the generator script fails on line 26 of half-pipe / lib / generators / half_pipe / install_generator.rb

**Comment by joefiorini** [comment]
Good call. I'm not sure how to best handle that, seems like we'd have to do one-off checks and code for every template engine. Unless anyone else knows of a better way?

**Comment by joefiorini** [comment]
I think we can use some of @plataformatec's generators as examples. Simple Form seems to do this by including a few different types of templates in their generator (see https://github.com/plataformatec/simple_form/tree/master/lib/generators/simple_form/templates).

**Comment by dmpawlowski** [comment]
Coming out of left field. You said this might be a good spot for me to help on half-pipe.  What files would we need to create haml duplicates for?  Anything with erb?

Don

**Comment by joefiorini** [comment]
Yeah, I think that's it, just erb files.

**Comment by cj** [comment]
I see this has the label ready, how would you use it?  Right now at the latest master I get `Warning: An error occurred while processing a template (Unexpected token %=). Use --force to continue.` when running `grunt server` I'm using haml for all my templates.

**Comment by matthewbivins** [comment]
I'm also having a problem with this...getting the error upon installing (rails g half_pipe:install) that I don't have an application.html.erb (because mine is haml). I'd love to start using half-pipe, though!
